### PR TITLE
Cleanup top-level directory

### PR DESCRIPTION
**Pull request by tessr**
Our top-level directory has been getting very long, and I'd like to clean it up. Here's an attempt at a start - it's possible we shouldn't delete or move these things, but if we determine that anything is load-bearing, we can always bring it back. 

* Mergify, Markdownlint: Can live in the .github directory, moved
* Philosophy.md: Was not really used or adhered to 
* build.sh: Moved into scripts 
* Appveyor: Not sure we were using this at all? 
* Codecov: There's another codecov config in the .github directory already; do we need both? 

Next: Perhaps consolidating the docker configs and/or makefiles? Moving some of the top-level markdown files into /docs?

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -1,27 +0,0 @@
-coverage:
```
Comment (marbar3778) [inline codecov.yml:1]: we should keep this.

Comment (tessr) [reply codecov.yml:1]: So this was one of my questions--why is there _also_ a codecov.yml file in the .github directory?

Comment (marbar3778) [reply codecov.yml:1]: oh now I see the question in the description. I don't see a codecov.yml file in the github dir (?).

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -27,6 +27,6 @@ jobs:
           DEFAULT_BRANCH: master
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           VALIDATE_MD: true
-          MARKDOWN_CONFIG_FILE: .markdownlint.yml
+          MARKDOWN_CONFIG_FILE: .github/markdownlint.yml
```
Comment (marbar3778) [inline .github/workflows/linter.yml:30]: nit: if we move `markdownlint.yml` into a `linter` folder in `.github/` we can remove this line.

Comment (tessr) [reply .github/workflows/linter.yml:30]: Ooh, even better!

File: codecov.yml (hunk 1 above)
Comment (tessr) [reply codecov.yml:1]: Oh wait. You're totally right, will add it back. This is what I get for doing these things in the middle of the night!

Comment (tessr) [reply codecov.yml:1]: Oh, but I _can_ move it to `.github/`: https://github.com/cosmos/cosmos-sdk/pull/7828#discussion_r518337441 🎉

**Review by marbar3778 — APPROVED** [review]
LGTM, love this cleanup!!! 🥳

**Comment by tessr** [comment]
Thank you for being eagle-eyed in your review!

**Comment by marbar3778** [comment]
ahh moving build.sh broke the reproducible build script. But I would say this is more a problem how the docker container assumes the script is in the root. Will look into fixing this on Monday

**Comment by tessr** [comment]
Ah, good catch. I'll take a look right now

**Pull request merged**